### ISSUE TITLE
No exception raised them running `Complex('a+5i')`

### ISSUE THREAD

**Issue by ShockwaveNN** [oracle/truffleruby#2765]
Hi, trying to migrate some of my libraries to truffleruby and found this difference between truffleruby and MRI

The code is quite simple:
```
Complex('a+5i')
```

Running this (in `irb` for example) on MRI 3.1 raise an exception:
```
(irb):3:in `Complex': invalid value for convert(): "a+5i" (ArgumentError)
```

Running this in `truffleruby 22.2.0`returns `(0+0i)` without any exception

**Comment by andrykonchin** [comment]
Thank you for reporting the issue. Fixed in fb8887c8f85c9f3a57d66d38d2625022d4191556.

**Comment by ShockwaveNN** [comment]
Thanks
